### PR TITLE
Load and display uncommitted file in BlameEditor

### PR DESCRIPTION
**Pull request by stefanknotzer**
1) Load and display uncommitted file in BlameEditor
2) Simplifying the logic how to find the correct blame
3) Enable file context menu in DoubleTreeWidget to navigate revisions, ... (Right click at any position of the DoubleTreeWidget, but not in the editor and not in the treeview)

### CODE REVIEW

File: src/ui/DoubleTreeWidget.cpp [hunk 1]
```diff
@@ -377,32 +423,15 @@ void DoubleTreeWidget::loadEditorContent(const QModelIndex &index)
...
-        found = true;
-        break;
-      }
-    }
-    if (!found)
-      break;
-  }
+  RepoView *view = RepoView::parentView(this);
+  git::Blob blob = mDiff.isValid() ? view->repo().lookupBlob(mDiff.id(mDiff.indexOf(name), git::Diff::NewFile)) :
+                                     git::Blob();
```
Comment (stefanknotzer) [inline src/ui/DoubleTreeWidget.cpp:428]:
```suggestion
  int idx = mDiff.isValid() ? mDiff.indexOf(name) : -1;
  git::Blob blob = idx < 0 ? git::Blob() :
                             view->repo().lookupBlob(mDiff.id(idx, git::Diff::NewFile));
```

File: src/ui/DoubleTreeWidget.cpp [hunk 2]
```diff
@@ -377,32 +423,16 @@ void DoubleTreeWidget::loadEditorContent(const QModelIndex &index)
...
+  git::Blob blob = idx < 0 ? git::Blob() :
+                             view->repo().lookupBlob(mDiff.id(idx, git::Diff::NewFile));
+
+  mEditor->load(name, blob, commit);
 
-  if (found)
-    mEditor->load(name, blob, commit);
+  // Disable blame editor when an uncommitted file is displayed
+  if (mDiff.isValid())
+    mEditor->editor()->setEnabled(!mDiff.isStatusDiff());
```
Comment (stefanknotzer) [inline src/ui/DoubleTreeWidget.cpp:435]:
```suggestion
```

**Review by exactly-one-kas — APPROVED** [review]

**Comment by Murmele** [comment]
@exactly-one-kas 
I'm fine with it, if you are, you can merge.

**Comment by Murmele** [comment]
@stefanknotzer can you rebase?

**Comment by stefanknotzer** [comment]
@Murmele I see "This branch has no conflicts with the base branch"...
Am I missing something?

**Comment by Murmele** [comment]
It showed me previously, don't know why

**Pull request merged**